lib/update/reconciler.go

```diff
@@ -84,22 +87,28 @@ func (r *reconciler) trySyncChangelogFromEtcd(ctx context.Context) error {
 	}
 
 	if shouldSync {
-		return trace.Wrap(r.syncChangelog(r.backend, r.localBackend))
+		return trace.Wrap(r.syncChangelog(ctx, r.backend, r.localBackend))
 	}
 
 	return nil
 }
 
 // syncChangelog will sync changelog entries from src to dst storage
-func (r *reconciler) syncChangelog(src storage.Backend, dst storage.Backend) error {
-	return trace.Wrap(SyncChangelog(src, dst, r.cluster, r.operationID))
+func (r *reconciler) syncChangelog(ctx context.Context, src, dst storage.Backend) error {
```

Comment:
I've wrapped the query in a retry loop but left the original precondition that skips the reads if the cluster is not healthy. 

Initially I tried to work it around by first introducing a stricter check for when to skip the reads by testing for the etcd service being masked (which is the case once it has been shutdown on the node) but it does not work in a multi-node setting until all etcd instances have been shut down.
I then attempted to explicitly skip plan syncs by configuring it as an attribute of the plan step and checking it in the engine but it required a lot of changes and I failed to come up with a way to pair it with the FSM as it would require FSM to know about the reconciler and be able to provide custom handling during upgrades.
In the end, I figured it was all not worth the effort and the changes.